src/lazy/decoder.rs

```diff
@@ -257,6 +258,21 @@ pub enum LazyRawFieldExpr<'top, D: Decoder> {
 }
 
 impl<'top, D: Decoder> LazyRawFieldExpr<'top, D> {
+    pub fn resolve(self, context: EncodingContextRef<'top>) -> IonResult<FieldExpr<'top, D>> {
+        use LazyRawFieldExpr::*;
+        let field = match self {
+            NameValue(name, value) => FieldExpr::NameValue(
+                name.resolve(context),
+                LazyExpandedValue::from_literal(context, value),
+            ),
+            NameEExp(name, eexp) => {
+                FieldExpr::NameMacro(name.resolve(context), eexp.resolve(context)?.into())
+            }
+            EExp(eexp) => FieldExpr::EExp(eexp.resolve(context)?),
+        };
+        Ok(field)
+    }
```

Comment:
🪧 A convenience method pattern used widely throughout the crate; `resolve` turns a raw value (or in this case, field) into its higher-level, context-ful equivalent.